Week 3/Solution.java

```diff
@@ -0,0 +1,64 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+    public List<List<Integer>> threeSum(int[] n) {
+        List<List<Integer>> res = new  ArrayList<List<Integer>>();
+        if(n == null || n.length<3) return res;
+        int len = n.length;
+        HashSet<Integer> checked = new HashSet<Integer>();
+        for(int i = 0; i<len; i++){
+            int goal = n[i]*-1;
```

Comment:
Thanks! I'll try to add more comments because it's much easier to understand the code 😄